src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Setup/DotNetEnvironment.cs

```diff
@@ -0,0 +1,123 @@
+using Microsoft.VisualStudio.Threading;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Setup;
+
+/// <summary>
+/// Provides information about the .NET environment and installed SDKs by querying the Windows registry.
+/// </summary>
+[Export(typeof(IDotNetEnvironment))]
+internal class DotNetEnvironment : IDotNetEnvironment
+{
+    private readonly IFileSystem _fileSystem;
+    private readonly IRegistry _registry;
+    private readonly IEnvironment _environment;
+
+    [ImportingConstructor]
+    public DotNetEnvironment(IFileSystem fileSystem, IRegistry registry, IEnvironment environment)
+    {
+        _fileSystem = fileSystem;
+        _registry = registry;
+        _environment = environment;
+    }
+
+    /// <inheritdoc/>
+    public Task<bool> IsSdkInstalledAsync(string sdkVersion)
+    {
+        try
+        {
+            string archSubKey = GetArchitectureSubKey(_environment.ProcessArchitecture);
+            string registryKey = $@"SOFTWARE\dotnet\Setup\InstalledVersions\{archSubKey}\sdk";
+
+            // Get all value names from the sdk subkey
+            string[] installedVersions = _registry.GetValueNames(
+                Win32.RegistryHive.LocalMachine,
+                Win32.RegistryView.Registry32,
+                registryKey);
+
+            // Check if the requested SDK version is in the list
+            foreach (string installedVersion in installedVersions)
+            {
+                if (string.Equals(installedVersion, sdkVersion, StringComparison.OrdinalIgnoreCase))
+                {
+                    return Task.FromResult(true);
+                }
+            }
+
+            return Task.FromResult(false);
+        }
+        catch
+        {
+            // If we fail to check, assume the SDK is not installed
+            return Task.FromResult(false);
+        }
+    }
+
+    /// <inheritdoc/>
+    public string? GetDotNetHostPath()
+    {
+        // First check the registry
+        string archSubKey = GetArchitectureSubKey(_environment.ProcessArchitecture);
+        string registryKey = $@"SOFTWARE\dotnet\Setup\InstalledVersions\{archSubKey}";
+
+        string? installLocation = _registry.GetValue(
+            Win32.RegistryHive.LocalMachine,
+            Win32.RegistryView.Registry32,
+            registryKey,
+            "InstallLocation");
+
+        if (!string.IsNullOrEmpty(installLocation))
+        {
+            string dotnetExePath = Path.Combine(installLocation, "dotnet.exe");
+            if (_fileSystem.FileExists(dotnetExePath))
+            {
+                return dotnetExePath;
+            }
+        }
+
+        // Fallback to Program Files
+        string programFiles = _environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+        string dotnetPath = Path.Combine(programFiles, "dotnet", "dotnet.exe");
+        
+        if (_fileSystem.FileExists(dotnetPath))
+        {
+            return dotnetPath;
+        }
```

Comment:
Here's an example where not returning `null` from `GetFolderPath` could silently do something unexpected. While `ProgramFiles` is unlikely to not exist, if `GetFolderPath` returned `""` then we could accidentally pick up a file relative to the current directory. I like returning `null` because it forces consuming code to handle the failure case. I understand that it's just wrapping the BCL, but I think we can improve the contract a bit.